### DISCUSSION TITLE
What types of icons should be included in devicon?

### DISCUSSION THREAD

**Discussion by Thomas-Boi** [devicons/devicon#353] [General]
Hello guys,

We would like to ask all of you to help us narrow down the goals of Devicon. Currently, we think that the definition is too broad and sometimes, we don't know whether an icon belongs to our repo.

Here is Devicon's current definition:
> Devicon aims to gather all logos representing **development languages and tools**. 

Also, our name Devicon is "dev icon" shorterned together :). Basically, this means that we want icons that focuses on technological development languages and tools. 

*For languages, these includes but are not limited to:*
* Programming languages (Java, Python, C, . . .)
* Framework/libraries based on the above languages (React, CakePHP, . . .)
* Databases (MySQL, CouchDB, . . .)

*For tools, these includes but are not limited to:*
* Package managers (npm, bower . . .)
* Operating Systems (Windows, *nix, . . .)
* IDE or code editor (Pycharm, Codepen . . .)

___

The above examples are some cases where the icons clearly belongs to our repo. However, there are some categories where we don't know for sure, such as:

* Tools that are indirectly related to tech development (Slack, Trello, Jira, Adobe products, . . .)
* Tech companies (Facebook, Twitter, Apple, . . .)
* Job searching platforms or social media (LinkedIn, HackerRank, . . .)

**What we want to discuss with you guys:**

1. Should we accept tools that are indirectly related to tech development? While we already have some icons in our repo (Slack, Photoshop, Trello, . . .), do you think that they belongs here? Should we continue accepting them?

2. If we do accept tools that are indirectly related to tech development, how indirect should the connection be before we stop accepting it? Microsoft Word or Google Calendar can be used for tech jobs, should we accept icons from these tools?

3. What count as tech companies? While Google, Facebook, Amazon, . . . are clear big tech companies, do Netflix, Uber, and AirBnB count? Should we even accept icons about these companies in the first place?

Those are the main questions that we can think of for now. If you guys have any questions you would like to bring up, please let us know and we can add them to the discussion list.

Looking forward to the discussion :)

**Comment by Thomas-Boi** [comment]
Here are my thoughts on the above questions:

> 1. Should we accept tools that are indirectly related to tech development? While we already have some icons in our repo (Slack, Photoshop, Trello, . . .), do you think that they belongs here? Should we continue accepting them?

Yes, I think we should. While programmers might not directly use these tools, we might use the products created by these tools. Also, a lot of tech jobs closely collaborate with "semi-tech" jobs, like web designers. I feel that these closely coupled jobs are a part of the tech industry too and their tools can be a part of Devicon.

For example, in web development, you might work with images that were created in Photoshop. Knowing Photoshop and other image manipulation apps are also considered a huge asset or even a requirement. Therefore, it makes sense to have Photoshop in Devicon.

> 2. If we do accept tools that are indirectly related to tech development, how indirect should the connection be before we stop accepting it? Microsoft Word or Google Calendar can be used for tech jobs, should we accept icons from these tools?

We should stop at more "general" products. Word and Google Calendar can be used by anyone from any industry, therefore they are not development tools. A good way to differentiate these products is to look at their customer base. If it's built specifically for tech jobs or for closely-coupled jobs in mind, we should accept it.

For example: 3D modelling apps and sound editing softwares can be seen as its own fields. These tools are built for 3D modelers and sound engineers in mind. While their products can be used in game development, the fields are not as closely coupled as web designers and web developer. Therefore, their icons should not go into Devicon.

I do admit sometimes this is tricky, such as Slack and LinkedIn. Therefore, we might have to take things on a case-by-case basis. If many people sees it as an an essential part of the tech industry (like Slack and LinkedIn), we can accept them. 

> 3. What count as tech companies? While Google, Facebook, Amazon, . . . are clear big tech companies, do Netflix, Uber, and AirBnB count? Should we even accept icons about these companies in the first place?

I think we should not have them in our repo. While these companies contribute a lot to the tech industry in terms of innovation, they are not tools or languages. However, if their name is also a development tool then we can accept them. So Google, the browser, should be accepted but Google, the company, should not. Facebook, the social media site, should not be accepted but its tools (React and GraphQL) should be. 

Of course, we can make exceptions and add companies to the repo if that's what our users want. However, they are so different from our current definition that they warrant their own repo.

**Comment by ezSaldana** [comment]
I've contributed with some icons and have been in this discussion before, in my opinion:

> 1. Should we accept tools that are indirectly related to tech development? While we already have some icons in our repo (Slack, Photoshop, Trello, . . .), do you think that they belongs here? Should we continue accepting them?

As a full stack web developer focused on front end, i think that at least the ones i've added, belongs to `Devicon`, because i've used them at least a couple of times in my job. As @Thomas-Boi mentions, it's tricky sometimes to know which ones accomplish with the goals of the brand, but as far as they are useful for tech creation or developement, i think they should be accepted.

> 2. If we do accept tools that are indirectly related to tech development, how indirect should the connection be before we stop accepting it? Microsoft Word or Google Calendar can be used for tech jobs, should we accept icons from these tools?

This is tricky as well cause you can programmatically create documents or templates that can help you to accomplish some tasks _i.e._ create a commercial `pdf or docx` invoice or use a `csv` document to import and work with data. They are general tools, maybe not created for developers in first instance, but they're useful in some cases.

At the same time, there are some tools that can provide you with functionalities that are related to a technology that might not be as related as some would want. In the case of google calendar, they provide an `API` that can be used in development, so i think these icons should be included.

Taking another example, the case of `matlab` that was created for students and is used by mathematicians and investigators, these people might not be developers or related to the area, but at the end it counts as a devtool.

> 3. What count as tech companies? While Google, Facebook, Amazon, . . . are clear big tech companies, do Netflix, Uber, and AirBnB count? Should we even accept icons about these companies in the first place?

I think that they should be accepted, because as i mentioned with the google calendar example, some of them provide tools that are related to the area. Think about when you sign up on a new page or ecommerce, you can find a google or facebook, even twitter signup. And in most cases if not in all, the branding is the same.

As more examples, facebook provides pixel that's used like G-Analytics or even the business part, digital marketing and more services that are used nowadays by many digital marketing agencies or ecommerce business as part of their job.

Companies you mention like Airbnb have developed a lot of packages or useful libraries used today like `Enzyme` and `Air Map view`, but at least these have their own branding ("logos")...

## Conclusion

I think that this is a wide topic and the use of **tools** in the definition gives a gateway to many possibilities. In my point of view, a tool is something that you can use to make your work easier and as soon as a product accomplish this, it fits in the definition.

You can add an indirect tools section to solve this, or a lab section just like material-ui, and depending on the use and acceptance of an icon, decide if it's worth it or not.

At the end, the decision of which icons someone can add will be yours.

Greetings 😃

**Comment by joshminnie** [comment]
My 2¢

> 1. Should we accept tools that are indirectly related to tech development? While we already have some icons in our repo (Slack, Photoshop, Trello, . . .), do you think that they belongs here? Should we continue accepting them?

I think that's a really hard line to draw, if you do accept them. There are other alternatives that are doing something similar to what devicons does (like [simpleicons](http://simpleicons.org/)).

> 2. If we do accept tools that are indirectly related to tech development, how indirect should the connection be before we stop accepting it? Microsoft Word or Google Calendar can be used for tech jobs, should we accept icons from these tools?

There are probably a lot more here than it's worth keeping up with, keep the focus narrow and you have more time to spend on what really matters to deliver quality (which by the way, I think you've done).

> 3. What count as tech companies? While Google, Facebook, Amazon, . . . are clear big tech companies, do Netflix, Uber, and AirBnB count? Should we even accept icons about these companies in the first place?

Again, there are plenty of sources for these types of icons. Stay focused, and you will deliver the quality that people need from you.

**EDIT**

Another point of view, playing my own "devil's advocate"...

You could always break those out into other repositories if you do so choose to keep them, giving weight and importance to them in a slightly different perspective.

**Comment by amacado** [comment]
I appreciate the discussion since this is a community driven project. As previous participants of this discussion pointed out it's pretty hard to draw the line since you could say that any tool you are using when you develop software (web, mobile, desktop, ,...) is a "software development tool". If you bring it to the extrem: A whiteboard in your office could be seen as a "related product for software development".

I think that devicon should accept icons which are directly related to **software development**. If a company (f.e. Google) decides to provide languages or tools for developers f.e. `Go` than `Go` is the icon we would add but not the company who provided that. In a perspective tools like Word or Excel sure can be used by a developer, but to be honest: You don't do software development in word. Sure you can write Plugins/Add-ons for Microsoft Office tools, but then I would say: Add the logo of the programming language you use for the Plugin and not Word itself. There are plenty of icon libs out there which have icons for Word and such common used tools.

In my opinion we should focus on tools and technologies which are made for software developers. Like in the origin description of this repository:

> Devicon aims to gather all logos representing development languages and tools  

To quote @Thomas-Boi:
>We should stop at more "general" products. Word and Google Calendar can be used by anyone from any industry, therefore they are not development tools. A good way to differentiate these products is to look at their customer base. **If it's built specifically for tech jobs or for closely-coupled jobs in mind, we should accept it**.
>For example: 3D modelling apps and sound editing softwares can be seen as its own fields. These tools are built for 3D modelers and sound engineers in mind. While their products can be used in game development, the fields are not as closely coupled as web designers and web developer. Therefore, their icons should not go into Devicon.

I would extend this description to not only web developers but software developers (which includes mobile, web, desktop). When taking about game development it's a little tricky but I admit to have a conversation on specific icon level (f.e. should we add Unity engine?)

**Comment by Thomas-Boi** [comment]
Hi all,

I'm very glad to see everyone's thought on this discussion so far. They are essential to Devicon and it helps maintainers like us to make this repo better for everybody :) 

So far, here are the things that stood out to me based on the current discussion:

* Programming languages of all kinds are OK. I think Matlab, R, and even Racket should belong in our repo (no disrespect to those languages).
* Tools relating to industries closely related to tech should be accepted. 
* Tools that are too general should not be added. Special cases can be treated on a case-by-base basis, such as Slack or LinkedIn
* Devicon does not need to have ALL tools. There are other similar websites that have icons of these tools that users can use. We should instead focus on quality and the fact that we provide fonts of these tools.
* Companies logo should not belong to repo UNLESS their logo is also their product's logo. Only the tools or languages created by those companies should be accepted. 

Regarding game engines, I think that we should accept them because:
* Game developers do code and use programming languages. Therefore, their industry belongs in the software development category. Disclaimer: we are looking at a game software developer specifically, not a jack-of-all-trade indie developer.
* Code on their own does nothing. Unity and Unreal Engine is the environment for the program to manipulate objects and scenes. Without the engines, the code would do nothing. This is similar to JavaScript and the browsers, which are in Devicon and fits the definition of tools (Chrome, Firefox, etc . . .)

That being said, game development is a slippery slope. This is a category that we will need to look at on a case-by-case basis.

So here is what we will do going forward:
* I'll leave this discussion up for about 1 more month. This is so we can get as much feedback if possible. 
* @amacado and I will review this discussion at that time and come up with a new guideline.
*  We'll edit the CONTRIBUTING.md so that it reflects the new guideline.

That is all, I look forward to see more comments :)

Cheers,
Thomas

**Comment by BenSouchet** [comment]
Hi,
Don't know if this discussion is still open, but I haven't find a proper list of what icons category can or cannot be added to Devicon.
So I will share my opinion as lists: Allowed and Not Allowed (I repeat this is only my opinion 🙂)

#### 1. The element you want to see added to **_Devicon_** must be at a minimum popular (because adding an icon for 2 people will increase the weight of the font unnecessarily).

#### 2. The element must fall into at least one of the category of the **_Allowed_** or **_Need Approbation_** section below:

## Allowed
- [Operating systems](https://en.wikipedia.org/wiki/Operating_system) (Linux, MacOS, Windows, ...)
- [Programming languages](https://en.wikipedia.org/wiki/List_of_programming_languages) (C, C++, Python, ...)
- [CLI languages](https://en.wikipedia.org/wiki/List_of_CLI_languages) (PowerShell, Swift, Visual Basic, ...)
- [Command-line interpreters](https://en.wikipedia.org/wiki/List_of_command-line_interpreters) (Unix Shell, zsh, bash, ...)
- [Compilers](https://en.wikipedia.org/wiki/List_of_compilers) (GCC, LLVM, Clang, ...)
- [IDEs](https://en.wikipedia.org/wiki/Comparison_of_integrated_development_environments) (Visual Studio, PyCharm, Xcode, Atom, ...)
- [Markup languages](https://en.wikipedia.org/wiki/List_of_document_markup_languages) (HTML, LaTeX, XML, Markdown, ...)
- [Style sheet languages](https://en.wikipedia.org/wiki/Style_sheet_language) (CSS)
- [Style sheet preprocessors](https://www.envato.com/blog/css-preprocessors/) (Less, Sass, SCSS, Stylus, ...)
- [Packages managers](https://en.wikipedia.org/wiki/List_of_software_package_management_systems) (npm, bower, Homebrew, NuGet, ...)
- [Database programs & systems]() (MangoDB, MySQL,SQLite, ...)
- [Database tools](https://en.wikipedia.org/wiki/Comparison_of_database_administration_tools) (phpMyAdmin, pgAdmin, ...)
- [Software Frameworks](https://en.wikipedia.org/wiki/Software_framework) (Electron, Qt, ...)
- [Web Frameworks](https://en.wikipedia.org/wiki/Comparison_of_server-side_web_frameworks) (Django, Angular, ...)
- [Runtime environments](https://en.wikipedia.org/wiki/Runtime_system) (Node.js, ...)
- [Web server softwares](https://en.wikipedia.org/wiki/Comparison_of_web_server_software) (apache, nginx, ...)
- [Source code hosting platforms](https://en.wikipedia.org/wiki/Comparison_of_source-code-hosting_facilities) (Github, Gitlab, Bitbucket, ...)
- [Testing tools](https://en.wikipedia.org/wiki/Software_testing) (Jasmine, )
- [Static Site Generators](https://en.wikipedia.org/wiki/Web_template_system#Static_site_generators) (Jekyll, Hugo, eleventy, ...)
- [Web browsers](https://en.wikipedia.org/wiki/List_of_web_browsers) (Google Chrome, Mozilla Firefox, Microsoft Edge, ...)
- [OS-level virtualization](https://en.wikipedia.org/wiki/OS-level_virtualization) (Docker, chroot, ...)
- [FTP Client Softwares](https://en.wikipedia.org/wiki/Comparison_of_FTP_client_software) (Filezilla, WinSCP, ...)
- Others popular modules, packages, libraries (Babel, Tensorflow, Jupyter, ...)

## Need Approbation
### For these categories we need also to see why it's related to development before accepting.
- Web Services for developers (gitter, Codepen, ...)
- [Text Editors](https://en.wikipedia.org/wiki/List_of_text_editors) (Vim, Emacs, ...)
- [Platforms as a service](https://en.wikipedia.org/wiki/Platform_as_a_service) (Heroku, Meteor, Amazon Web Services, Google App Engine, ...)
- [Content management systems (CMS)](https://en.wikipedia.org/wiki/List_of_content_management_systems) (Wordpress, Drupal, Prestashop, ...)
- [Game engines](https://en.wikipedia.org/wiki/List_of_game_engines) (Unity, blender, Maya, ...)
- [Raster graphics editors](https://en.wikipedia.org/wiki/Raster_graphics_editor) (Photoshop, GIMP, ...)
- [Prototyping tools](https://en.wikipedia.org/wiki/Comparison_of_vector_graphics_editors) (Figma, Adobe XD, Sketch, Pencil, ...)

## Not Allowed (not exhaustive)
- [Social Media platforms](https://en.wikipedia.org/wiki/List_of_social_networking_services) (Facebook, LinkedIn, Twitter, ...)
- [Email marketing Services](https://en.wikipedia.org/wiki/Email_marketing) (Mailchimp, ...)
- [CRM tools](https://en.wikipedia.org/wiki/Customer_relationship_management) (Salesforces, ...)
- [Project management tools](https://en.wikipedia.org/wiki/Project_management) (Jira, Monday, ClickUp)
- [Internet providers](https://en.wikipedia.org/wiki/Category:Internet_service_providers_by_country) (AT&T, Verizon, ...)
- [Global web hosting platforms](https://en.wikipedia.org/wiki/Category:Web_hosting) (Hostinger, Bluehost, SiteGround, ...)
- [General Website builders](https://en.wikipedia.org/wiki/Website_builder) (Wix, Squarespace, ...)
- [Computer-aided design tools](https://en.wikipedia.org/wiki/Computer-aided_design) (AutoCAD, Solidworks, SketchUp, ...)

For me defining categories simplifies the approval process for a new icon to the project, and it avoids the frustration of not knowing the "rules" for the users.

Let me know that you think 😊

**Reply by Thomas-Boi** [reply]
I agree with your classifications here. It is a bit difficult for us because we have icons that don't fit these rules from the origin of the repo. 

If you don't mind, can we copy these guidelines and add it to the wiki?

**Reply by BenSouchet** [reply]
Yes you can copy this list, modify or extend it 🙂

And maybe a new major version should be made in which the icons are reorganized with a least one main category.
Of course the current version will still be available but simple discontinued.

With this new major version there is the ability to rename old icons, remove totally unrelated icons (Facebook, Twitter, ...) so that the project will be focused again on icons related to the DEV theme.

**Reply by amacado** [reply]
A "cleanup" which removes non-matching icons could be a good idea to sustain the purpose of this project. We should discuss which icons should be removed in the next major release.

**Reply by BenSouchet** [reply]
Ah, and I didn't express my point of view on the position that should have Devicon, how to know if an element can be on Devicon ? You need to answer simple questions:

1. Can you write code with the "element" ? (languages, IDEs, CLIs, StyleSheets, Markup, web frameworks, ...)
2. Can you handle code with the "element" ? (Packages managers, Database programs, Web server softwares, ...)
3. Can you transform code with the "element" ? (Compilers, Style sheet preprocessors, ...)
4. Can you add your code to the "element" ? (Game engines, Source code hosting platforms, Platforms as a service, ...)
5. Can you test you code with the "element" ? (Testing tools, Web browsers, ...)

Before looking to fit an element into a category these questions really help to know if it is related to development 😊

For me justification like **_"it's okay because i've used them at least a couple of times in my job"_** isn't good since everyday in my job I use **Gmail** but it place isn't on Devicon.

**Reply by Snailedlt** [reply]
> Ah, and I didn't express my point of view on the position that should have Devicon, how to know if an element can be on Devicon ? You need to answer simple questions:
> 
> 1. Can you write code with the "element" ? (languages, IDEs, CLIs, StyleSheets, Markup, web frameworks, ...)
> 2. Can you handle code with the "element" ? (Packages managers, Database programs, Web server softwares, ...)
> 3. Can you transform code with the "element" ? (Compilers, Style sheet preprocessors, ...)
> 4. Can you add your code to the "element" ? (Game engines, Source code hosting platforms, Platforms as a service, ...)
> 5. Can you test you code with the "element" ? (Testing tools, Web browsers, ...)
> 
> Before looking to fit an element into a category these questions really help to know if it is related to development 😊
> 
> For me justification like **_"it's okay because i've used them at least a couple of times in my job"_** isn't good since everyday in my job I use **Gmail** but it place isn't on Devicon.

One more to add to this:
6. Can you analyze code with the "element" ? (Static code analyzers, Linters, Analytics tools)

**Comment by dmantilla9** [comment]
Hello Comunity I’m looking for the Talend logo (ETL tool) and do you know if it is in the repo or if it exists in another one?
